### DISCUSSION TITLE
Error Running Commands via SSH on IBM i: "Memory Fault - Core Dumped"

### DISCUSSION THREAD

**Discussion by T26-AR** [codefori/vscode-ibmi#2354] [Q&A]
Hello,

I’m encountering an issue when trying to execute commands on an IBM i system via SSH. Specifically, when I connect to the system and issue a command (e.g., liblist) using /QOpenSys/usr/bin/qsh, I receive the following error:

`.: echo "This should be the only text!"
{
    "code": 0,
    "signal": null,
    "stdout": "This should be the only text!",
    "stderr": ""
}

.: echo $HOME && cd && test -w $HOME
{
    "code": 0,
    "signal": null,
    "stdout": "/home/MYUSRID",
    "stderr": ""
}

/home/MYUSRID: /QOpenSys/usr/bin/qsh
liblist
{
    "code": 139,
    "signal": null,
    "stdout": "",
    "stderr": "bsh: 2875305 Memory fault - core dumped"
}

/home/MYUSRID: /QOpenSys/usr/bin/qsh
system  "CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')"
{
    "code": 139,
    "signal": null,
    "stdout": "",
    "stderr": "bsh: 2875307 Memory fault - core dumped"
}

/home/MYUSRID: [ -d "/tmp" ]
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/MYUSRID: echo $SHELL
{
    "code": 0,
    "signal": null,
    "stdout": "/QOpenSys/usr/bin/bsh",
    "stderr": ""
}

/home/MYUSRID: /QOpenSys/usr/bin/expr `/QOpenSys/usr/bin/getconf ARG_MAX` - `env|wc -c` - `env|wc -l` \* 4 - 2048
{
    "code": 139,
    "signal": null,
    "stdout": "",
    "stderr": "bsh: 2875311 Memory fault - core dumped\nbsh: 2875312 Memory fault - core dumped\nbsh: 2875313 Memory fault - core dumped\nbsh: 2875314 Memory fault - core dumped"
}

/home/MYUSRID: rm -rf /tmp/code4itempO_quipGYRZrBvnRBA2OBBI && mkdir -p /tmp/code4itempO_quipGYRZrBvnRBA2OBBI
{
    "code": 139,
    "signal": null,
    "stdout": "",
    "stderr": "bsh: 2875316 Memory fault - core dumped"
}

/home/MYUSRID: test -r /QIBM/ProdData/IBMiDebugService/package.json
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

`

The command results in a "Memory fault - core dumped" error. I’m using SSH to connect and issue commands on an IBM i system.

The library ILEDITOR exists on the system and is accessible without any issues.

Has anyone encountered this issue or have any ideas on how to resolve it? Any help or guidance would be greatly appreciated!

Thank you!

**Comment by sebjulliand** [comment]
Not really sure what's happening here, but I can some references to `bsh`.

You may want to check out these instructions to make `bash` your default shell: https://ibmi-oss-docs.readthedocs.io/en/latest/troubleshooting/SETTING_BASH.html

Code for i should have checked that for you, but it won't hurt to check again and make it the default.
